src/Microsoft.Graph.Core/Requests/DeltaResponseHandler.cs

```diff
@@ -146,49 +145,106 @@ private async Task<JObject> DiscoverChangedProperties(JObject responseItem)
         /// <param name="changes">The list of properties returned in the response.</param>
         /// <param name="parentName">The parent object of this changed object.</param>
         /// <returns></returns>
-        private async Task GetObjectProperties(JObject changedObject, JArray changes, string parentName = "")
+        private async Task GetObjectProperties(JsonElement changedObject, List<string> changes, string parentName = "")
         {
             if (parentName != string.Empty)
             {
                 parentName += ".";
             }
 
-            foreach (JProperty property in changedObject.Children())
+            foreach (var property in changedObject.EnumerateObject())
```

Comment:
what happens if changedObject is null?

Reply:
JsonElement is actually a struct and is actually not nullable(value type).